docs/guides/end-to-end-testing/migration/selenium-to-cypress.mdx

```diff
@@ -0,0 +1,431 @@
+
+##### GraphQL
+
+Given the increasing prominence of GraphQL, the community has responded accordingly. Validating the contents of GraphQL
+requests or responses is entirely feasible using Cypress's native intercept functionality. This plugin enhances both
+the stability and usability of the intercept feature: [Shopify/cypress-graphql](https://github.com/Shopify/cypress-graphql)
+
+## Migrating test cases with Cucumber
+
+If you're currently using a cucumber framework, this Cypress plugin can assist you in transitioning smoothly.
+Although it alters the conventional `describe` and `it` functionality, the adjustment process should be relatively straightforward.
+
+The main difference would be that instead of adding the whole flow into a single `it` section, you add the different steps into
+an `it` like section, where the `it` is replaced with the standard Gherkin syntax.
+
+The first snippet comprises plain Cypress code, whereas the second snippet demonstrates the cucumber version
+
+```javascript
+it("I should see a search bar", () => {
+  cy.visit("https://www.duckduckgo.com");
+  cy.get("input").should(
+    "have.attr",
+    "placeholder",
+    "Search the web without being tracked"
+  )
+```
+
+```javascript
+import { When, Then } from '@badeball/cypress-cucumber-preprocessor'
+
+When('I visit duckduckgo.com', () => {
+  cy.visit('https://www.duckduckgo.com')
+})
+
+Then('I should see a search bar', () => {
+  cy.get('input').should(
+    'have.attr',
+    'placeholder',
+    'Search the web without being tracked'
+  )
+})
+```
+
+<br />
+As evident, the actual code or logic remains unchanged. The difference lies in the
+structure of the spec file and the framework's architecture. [Run cucumber/gherkin-syntaxed
+specs with Cypress](https://github.com/badeball/cypress-cucumber-preprocessor)
+
+## Integration with CI/CD Pipeline
+
+### Integration with the current Pipeline
+
+Cypress can be integrated into any pipeline technology using its headless runner. This headless runner leverages
+the Electron browser, ensuring minimal adaptation of the current CI/CD. Given the widespread usage of Docker in
+most pipelines today, the new Cypress pipeline is no exception. Integration becomes effortless by employing a Node.js
+docker image or one provided by Cypress via Docker Hub. These Docker images, not only furnished by Cypress,
+include various versions of Chrome, Firefox, and Edge.
+
+Executing the following command initiates a headless run against the Electron browser. Upon completion, a succinct
+table is generated, detailing the amount of tests for each cy.js file, in addition to the failed, skipped and passed
+tests with the run time of that specific cy.js file. At the bottom of the table, a comprehensive summary displays the
+total number of tests run, the duration of the run, and the counts of failed, passed, and skipped tests.
+
+```bash
+Npx cypress run
```

Comment:
```suggestion
npx cypress run
```

Mixed case works on Windows (which is case-insensitive / case-preserving) but not on Linux (case-sensitive). The correct command is `npx`. (Depending on the package manager, yarn or pnpm could also be used here.)

Exactly how to run Cypress in CI/CD depends on the CI provider and there is detailed documentation available for this under https://docs.cypress.io/guides/continuous-integration/introduction.